### PR TITLE
fix(spanner): rubocop failing due to bundler vendor directory

### PR DESCRIPTION
**Pull request by jiren**
Add vendor dir to rubocop exclude list

- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [X] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.

closes: #8417

### CODE REVIEW

**Review by skuruppu — APPROVED** [review]

**Comment by aeroastro** [comment]
Thank you ⭐

**Comment by skuruppu** [comment]
@dazuma if we could get an approval, it will unblock us from merging this PR.

**Comment by skuruppu** [comment]
@quartzmo if you could please give us an approval, we can merge this. Thanks heaps.

**Comment by hengfengli** [comment]
@jiren Can you please rebase this PR? I can't re-run the CI test and would like to trigger the CI test again.

**Pull request merged**
